truffle/contracts/Load/LoadContract.sol

```diff
@@ -231,6 +276,9 @@ contract LoadContract is Ownable {
 
         shipment.state = Shipment.State.CREATED;
         shipment.shipper = msg.sender;
+        shipment.vaultUri = _vaultUri;
```

Comment:
Setting the `vaultUrl`, `vaultHash`, and `carrier` in this way will not emit the events that correspond to these values changing.  Can you change this to utilize the methods that already exist for setting these fields which should handle emitting the events?